nixos/modules/services/web-apps/immich.nix

```diff
@@ -0,0 +1,299 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.immich;
+  isPostgresUnixSocket = lib.hasPrefix "/" cfg.database.host;
+  postgresEnv =
+    if isPostgresUnixSocket then
+      { DB_URL = "socket://${cfg.database.host}?dbname=${cfg.database.name}"; }
+    else
+      {
+        DB_HOSTNAME = cfg.database.host;
+        DB_PORT = toString cfg.database.port;
+        DB_DATABASE_NAME = cfg.database.name;
+        DB_USERNAME = cfg.database.user;
+      };
+  isRedisUnixSocket = lib.hasPrefix "/" cfg.redis.host;
+  redisEnv =
+    if isRedisUnixSocket then
+      { REDIS_SOCKET = cfg.redis.host; }
+    else
+      {
+        REDIS_PORT = toString cfg.redis.port;
+        REDIS_HOSTNAME = cfg.redis.host;
+      };
+
+  commonServiceConfig = {
+    Type = "simple";
+    Restart = "on-failure";
+    RestartSec = 3;
+
+    # Hardening
+    CapabilityBoundingSet = "";
+    NoNewPrivileges = true;
+    PrivateUsers = true;
+    PrivateTmp = true;
+    PrivateDevices = true;
+    PrivateMounts = true;
+    ProtectClock = true;
+    ProtectControlGroups = true;
+    ProtectHome = true;
+    ProtectHostname = true;
+    ProtectKernelLogs = true;
+    ProtectKernelModules = true;
+    ProtectKernelTunables = true;
+    RestrictAddressFamilies = [
+      "AF_INET"
+      "AF_INET6"
+      "AF_UNIX"
+    ];
+    RestrictNamespaces = true;
+    RestrictRealtime = true;
+    RestrictSUIDSGID = true;
+  };
+  inherit (lib)
+    types
+    mkIf
+    mkOption
+    mkEnableOption
+    ;
+in
+{
+  options.services.immich = {
+    enable = mkEnableOption "Immich";
+    package = lib.mkPackageOption pkgs "immich" { };
+
+    mediaLocation = mkOption {
+      type = types.path;
+      default = "/var/lib/immich";
+      description = "Directory used to store media files. If it is not the default, the directory has to be created manually such that the immich user is able to read and write to it.";
```

Comment:
Does immich allow this directory to be read-only?

Reply:
No